pkg/engine2/path_selection/path_expansion.go

```diff
@@ -274,6 +274,20 @@ func ExpandPath(
 	for i, node := range nonBoundaryResources {
 		candidates[i] = make(map[construct.ResourceId]int)
 		candidates[i][node] = 0
+		resource, err := input.TempGraph.Vertex(node)
+		if err != nil {
+			errs = errors.Join(errs, err)
+			continue
+		}
+		// we know phantoms are always valid, but we want to ensure we make them valid based on src and target validity checks
+		valid, err := checkCandidatesValidity(ctx, resource, path, input.Classification)
+		if err != nil {
+			errs = errors.Join(errs, err)
+			continue
+		}
+		if !valid {
+			candidates[i][node] = -1000
```

Comment:
If it's not valid, shouldn't it just be removed? Not-valid should be the same as not-preferred.

Reply:
will reword. all phantoms are not inherently valid, but can be made valid. So we want to make sure they are weighted less than a valid reusable candidates. Probably should rename the method since its not actually marking things as invalid yet, just as helping determine correct weights